rafx-base/src/owned_pool.rs

```diff
@@ -0,0 +1,91 @@
+use crossbeam_channel::{bounded, Receiver, Sender};
+use std::ops::{Deref, DerefMut};
+
+pub struct OwnedPool<T> {
```

Comment:
I feel like we already had something like this but it was probably GPU-specific and waiting N frames until clearing a resource and allowing them to be reused. Maybe it was removed.

Reply:
Yes, the `PooledResourceAllocator`. We still have it & you identified the difference -- it waits `N` frames until the resource is cleared & allowed to be reused. Also, I'm not sure it is actually `pub` outside of the `rafx` code that uses it. The function signature for creating a new resource is also different -- it assumes the resource is created by a `RafxDeviceContext`.